ioSPI/osf_upload.py

```diff
@@ -0,0 +1,174 @@
+"""Module to upload data to OSF.io generated by simSPI TEM Simulator."""
+import typing
+from pathlib import Path
+
+import requests
+
+
+class TEMUpload:
+    """Class to upload data to OSF.io generated by simSPI TEM Simulator.
```

Comment:
The ioSPI module is agnotic from the TEM simulator (and any type of simulator).

Either:
- put only functions that know how to upload to OSF (independently of where the data initially came from)
- if this is TEM-specific, it should go in simSPI

Reply:
Pretty much this whole file is already simSPI agnostic. I think you can just change some docstrings and function / class names for generality. I didn't notice much code at all that was intimately connected to the type of data that will be uploaded.